src/Metadata/PersistenceMeansInterface.php

```diff
@@ -0,0 +1,22 @@
+<?php
+
+/*
+ * This file is part of the API Platform project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace ApiPlatform\Metadata;
+
+/**
+ * Marker interface to be implemented by value objects encapsulating data needed by {@see ProviderInterface providers}
+ * to retrieve resources from a persistence layer.
+ */
+interface PersistenceMeansInterface
```

Comment:
IMHO we must not use `persistence` in the name, as we tried to avoid it for v3 (the processor can persist, but also send mails, do a computation...).

Maybe:

```suggestion
namespace ApiPlatform\State;

/**
 * Marker interface to be implemented by value objects encapsulating data needed by {@see ProcessorInterface processors} and {@see ProviderInterface providers}.
 */
interface OptionsInterface
```

Reply:
I really like the name for state options

Reply:
The goal of the interface is described in its PHPDoc. It makes no sense for a provider which sends mail to use it. Is only purpose is about persistence and its name should reflect it.

Reply:
Why? API Platform doesn't care about what the processor does (except in the bridges obviously).